### DISCUSSION TITLE
Deploying on Cloud

### DISCUSSION THREAD

**Discussion by jordimassaguerpla** [Project-MONAI/monai-deploy-workflow-manager#641] [General]
Hi!

I would like to try Monai Deploy on Cloud to test some of the models in the Model Zoo.

I see I need to deploy several components: orthanc, monai deploy workflow manager, monai deploy informatics gateway, rabbit message queue, mongodb, ohif dicom viewer (do I miss anything?).

I've seen those helm charts in [the deploy folder](https://github.com/Project-MONAI/monai-deploy-workflow-manager/tree/develop/deploy/helm)  and also those setups for [performance testing](https://github.com/Project-MONAI/monai-deploy/tree/main/performance-testing) plus the [demo](https://github.com/Project-MONAI/monai-deploy/tree/main/demos/ig-tm-map) .

However, it is a lot to digest.

Can you give me some hints/advice ? Can I deploy on any cloud or should I better do it on AWS EKS service? I read that only aws storage is supported, is this correct? Should I start with the deploy folder](https://github.com/Project-MONAI/monai-deploy-workflow-manager/tree/develop/deploy/helm)  and then implement a helm chart for orthanc?

Thanks in advance and congratulations on this great project!!

**Comment by mocsharp** [comment]
Hi @jordimassaguerpla, 

To run MONAI Deploy, you will need Workflow Manager, Informatics Gateway, MongoDB, RabbitMQ, and MinIO.

One option to test out all components is to use [MONAI Deploy Express](https://github.com/Project-MONAI/monai-deploy/tree/main/deploy/monai-deploy-express) which includes a docker-compose file that brings up all required + optional components. This option allows you to deploy to any system and cloud provider quickly.

For helm chart deployment, I think @neildsouth, @woodheadio, or @nakfour may be able to help.

**Comment by neildsouth** [comment]
I would be able to help out with the helm stuff if you need argo. I have
also checked the documentation here
https://github.com/Project-MONAI/monai-deploy-workflow-manager/blob/develop/guidelines/mwm-developer-setup.md
and it works as expected.

cheers
Neil


On Wed, 4 Jan 2023 at 16:04, Victor Chang ***@***.***> wrote:

> Hi @jordimassaguerpla <https://github.com/jordimassaguerpla>,
>
> To run MONAI Deploy, you will need Workflow Manager, Informatics Gateway,
> MongoDB, RabbitMQ, and MinIO.
>
> One option to test out all components is to use MONAI Deploy Express
> <https://github.com/Project-MONAI/monai-deploy/tree/main/deploy/monai-deploy-express>
> which includes a docker-compose file that brings up all required + optional
> components. This option allows you to deploy to any system and cloud
> provider quickly.
>
> For helm chart deployment, I think @neildsouth
> <https://github.com/neildsouth> or @woodheadio
> <https://github.com/woodheadio> may be able to help.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Project-MONAI/monai-deploy-workflow-manager/discussions/641#discussioncomment-4593228>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AY7534DNBUZLZKRIJH6KQR3WQWNR5ANCNFSM6AAAAAATQUSBZ4>
> .
> You are receiving this because you were mentioned.Message ID:
> <Project-MONAI/monai-deploy-workflow-manager/repo-discussions/641/comments/4593228
> @github.com>
>

-- 
The contents of this e-mail are intended only for the named addressees and 
may contain confidential and/or privileged material. If received in error, 
please contact us immediately on +44 (0)113 201 0600 and then delete the 
entire e-mail from your system. Unauthorised review, distribution, 
disclosure or other use of this information could constitute a breach of 
confidence. Your co-operation in this matter is greatly appreciated.

Answer Digital Ltd (reg. no. 03655429) 
is a company registered in England 
and Wales with registered office at 
Union Mills, 9 Dewsbury Road, Leeds 
LS11 5DD.

**Comment by dbericat** [comment]
I would totally try to use MONAI Deploy Express first if you just want to get started. https://github.com/Project-MONAI/monai-deploy/tree/main/deploy/monai-deploy-express

You have 2 options:

1. Take one of the 2 example MONAI Applications (MAPs) included within MONAI Deploy Express. Liver segmentation or Lung segmentation.
2. Take a model from Model Zoo, create an application (MAP) with MONAI Deploy App SDK (https://github.com/Project-MONAI/monai-deploy-app-sdk), and deploy it and run it on MONAI Deploy Express. 

If you want to run them on Kubernetes for some reason, I would wait until @woodheadio releases the new Helm charts part of Workflow Manager 0.2 that is coming soon. 

Any questions Jordi, please let us know! Adeu! :)

**Reply by jordimassaguerpla** [reply]
Hi @dbericat ! I want to try some example but I am getting very confused on which dcm to use with which argo and clinical workflows.

Can you point me to a specific example of a argo workflow with a clinical workflow with a dcm image? I was looking at the ones in monai deploy express but could not find the dcm images.

Gracies! :)

**Reply by jordimassaguerpla** [reply]
Sorry :) I am answering myself. There are links to some sample data at https://github.com/Project-MONAI/monai-deploy/tree/main/deploy/monai-deploy-express . I will use that :)

**Comment by dbericat** [comment]
Yes! use those! :)

**Comment by jordimassaguerpla** [comment]
Thanks @dbericat . See, this is what I am getting after all:

![image](https://user-images.githubusercontent.com/1148215/217629886-18aa7a61-bf44-496c-88de-5d47fa73f732.png)

Is this what it is expected?

It would be great to know what the segmentation model is expected to produce with a specific example, to know I have not missed any parameter or configuration.

Thanks

jordi

**Reply by dbericat** [reply]
I don't think we included the output to compare results are the same, did we @mocsharp @JonnyHancox?
